sklearn/metrics/_pairwise_distances_reduction/_argkmin.pxd.tp

```diff
@@ -0,0 +1,50 @@
+{{py:
+
+implementation_specific_values = [
+    # Values are the following ones:
+    #
+    #       name_suffix, INPUT_DTYPE_t, INPUT_DTYPE
+    #
+    # We also use the float64 dtype and C-type names as defined in
+    # `sklearn.utils._typedefs` to maintain consistency.
+    #
+    ('64', 'DTYPE_t', 'DTYPE'),
```

Comment:
**Not actionable**: I find the asymmetry with `cnp.float32_t` and `DTYPE_t` a bit weird. I prefer to have used `cnp.float64_t` everywhere and remove `DTYPE_t` all together. For someone new to the code base, it's easier to know what `cnp.float64_t` is at a glance. 

BTW, I would even go as far as using `float` and `double`. This is what NumPy typedefs `float*_t` to: [link1](https://github.com/numpy/numpy/blob/f1e50253a3b56375e795ee72032cca90a3832cd4/numpy/__init__.pxd#L285-L286) and [link2](https://github.com/numpy/numpy/blob/f1e50253a3b56375e795ee72032cca90a3832cd4/numpy/__init__.pxd#L703-L704). This way we do not even need to compile against NumPy.

That being said, I do not want to block this PR with this comment. I am okay with the `DTYPE_t` for now.

Reply:
> Not actionable: I find the asymmetry with cnp.float32_t and DTYPE_t a bit weird. I prefer to have used cnp.float64_t everywhere and remove DTYPE_t all together. For someone new to the code base, it's easier to know what cnp.float64_t is at a glance.

I used this for consistency with previous usage of Tempita in the code-base, but this is definitely hard to understand. I will use `cnp.float64_t` instead of `DTYPE_t` and `np.float64` instead of `DTYPE`.

> BTW, I would even go as far as using float and double. This is what NumPy typedefs float*_t to: link1 and link2. This way we do not even need to compile against NumPy.
> 
> That being said, I do not want to block this PR with this comment. I am okay with the DTYPE_t for now.

I agree that Cython implementations mixed a lot of a C-types and dtypes zoo, and would do follow-up PRs to unify it progressively across the whole code-base.

Reply:
> That being said, I do not want to block this PR with this comment. I am okay with the DTYPE_t for now.

Are you referring to the whole comment, or solely to its second part, namely

> BTW, I would even go as far as using float and double. This is what NumPy typedefs float*_t to: [link1](https://github.com/numpy/numpy/blob/f1e50253a3b56375e795ee72032cca90a3832cd4/numpy/__init__.pxd#L285-L286) and [link2](https://github.com/numpy/numpy/blob/f1e50253a3b56375e795ee72032cca90a3832cd4/numpy/__init__.pxd#L703-L704). This way we do not even need to compile against NumPy.

?

So as to know if I need to perform some renaming in this PR for the templating or if it would enlarge the scope of this PR too much.

Reply:
> if it would enlarge the scope of this PR too much.

It would increase the scope too much. I meant that the whole comment is not actionable in terms of this PR.